.github/workflows/ci_windows.yml

```diff
@@ -0,0 +1,52 @@
+# Copyright 2024 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+name: CI - Windows
```

Comment:
for consistency it'd be useful to have host arch in the CI names of everything (we're going to want windows_arm64 and then this is windows_x64 or whatever)

Reply:
SG. Looking through https://github.com/Tencent/ncnn/tree/master/.github/workflows they also put the toolchain (when gcc vs clang).

Reply:
> `build_test_all_macos_arm64` (no longer exists, this got confusing as we moved away from self-hosted mac runners)

https://docs.github.com/en/actions/using-jobs/choosing-the-runner-for-a-job#standard-github-hosted-runners-for-public-repositories
![image](https://github.com/user-attachments/assets/9c54509d-a4ca-4949-9c9e-1b40953a2b5f)


Am I reading this right?

`macos-13` is x86, while `macos-14` is arm (since these have M1 CPUs)